LayoutTests/imported/w3c/web-platform-tests/compression/decompression-bad-chunks.tentative.any.js

```diff
@@ -56,8 +56,8 @@ for (const chunk of badChunks) {
     const writer = ds.writable.getWriter();
     const writePromise = writer.write(chunk.value);
     const readPromise = reader.read();
-    await promise_rejects_js(t, TypeError, writePromise, 'write should reject');
-    await promise_rejects_js(t, TypeError, readPromise, 'read should reject');
+    await promise_rejects_js(t, TypeError, writer.close(), 'write should reject');
+    await promise_rejects_js(t, TypeError, reader.closed, 'read should reject');
```

Comment:
It is suspicious that we need to change the test while we are only switching from zlib to Apple Compression framework.
It might be due to lib eagerly doing some work and failing early while Apple Compression format is lazily doing things.
Can we understand what is going on here? Is there a way to use Apple Compression framework while keeping the exact zlib behavior with regards to error reporting?

Reply:
What is happening here that the invalid gzip bytes is throwing an error early in Zlib during the decompress stage. The compression framework on the other hand waits to error until we do the flush. Based on internal discussions with the Compression Framework team they did not indicate this was an issue.

We should not be tying this api exclusively to the Zlib behavior though. If compression framework decides to throw the error later we should support it.

Reply:
This seems ok, we can always revisit if there are too many behavioural differences.
Please add some comments explaining why you are changing the test though in the commit message.
Please also do a PR on the WPT repository with those changes as otherwise, these changes would get lost next time we resync WPT tests.

Reply:
@stwrt This change is probably fine, if you follow @youennf's guidance to update WPT. But I want to comment about how we frame a discussion like this.

Talking to the Compression Framework team is of course a good practice. But when we make decisions about the specifics of WebKit's behavior, our rationale should be based on what the web standard says, what other browsers do, and what web developers might reasonably expect, rather than what happens by default in our underlying frameworks. If an underlying framework does not match standards and/or other browsers and/or web developer expectations by default, it's our job to write the glue code that bridges that gap. If an underlying framework does not match standards and/or other browsers and/or web developer expectations for a Very Good Reason (TM), it's our job to influence change in standards and other browsers to conform to that Very Good Reason (TM).

Reply:
I'll make sure to add a comment and post the change to WPT.